extensions/ql-vscode/gulpfile.ts/typescript.ts

```diff
@@ -35,20 +35,35 @@ export function cleanOutput() {
     : Promise.resolve();
 }
 
-export function compileTypeScript() {
-  return tsProject
-    .src()
-    .pipe(init())
-    .pipe(tsProject(goodReporter()))
+export function compileEsbuild() {
+  return src("./src/extension.ts")
     .pipe(
-      write(".", {
-        includeContent: false,
-        sourceRoot: ".",
+      esbuild({
+        outfile: "extension.js",
+        bundle: true,
+        external: ["vscode", "fsevents"],
+        format: "cjs",
+        platform: "node",
+        target: "es2020",
+        sourcemap: "linked",
+        loader: {
+          ".node": "copy",
+        },
       }),
     )
     .pipe(dest("out"));
 }
 
-export function watchTypeScript() {
-  watch("src/**/*.ts", compileTypeScript);
+export function watchEsbuild() {
+  watch("src/**/*.ts", compileEsbuild);
+}
+
+export function checkTypeScript() {
+  // This doesn't actually output the TypeScript files, it just
```

Comment:
Can you explain a bit more why we're doing this?

Reply:
Sure! When ESBuild creates the bundle, it will strip all type information without checking any of it and only use the JavaScript which you'd get for bundling. This adds back TypeScript type checking to the build process. Without this, there would be no TypeScript checks in the build process and you could compile invalid TypeScript code without errors or warnings. We can remove this step since we already have `npm run check-types`, but that would mean you can bundle invalid code and need to explicitly run this separate command (although you'll probably also already get the errors in VSCode/IDE).

Reply:
Thanks. I guess I wasn't expecting ESBuild to not be doing type checking 🤷‍♀️